impl/src/main/java/com/sun/faces/facelets/compiler/TextUnit.java

```diff
@@ -285,7 +285,7 @@ private static String trimRight(String s) {
             if (Character.isWhitespace(s.charAt(i))) {
                 i--;
             } else {
-                return s;
+                return s.substring(0,i+1);
```

Comment:
Nice catch! This is indeed an existing bug.

Reply:
thanks ;)